File: hex_sum_gcd.jl

function hex_sum(value)
    HEXBASE = 16
    count = 0
    i = value
    while i > 0
        division, remainder = divrem(i, HEXBASE)
        count += remainder
        i = division
    end
    return count
end

function solve_range(lower, upper)
    countnon1gcds = 0
    for i in lower:upper
        f_i = hex_sum(i)
        hex_gcd = gcd(i, f_i)
        if hex_gcd != 1
            countnon1gcds += 1
        end
    end

    println(countnon1gcds)
end

function main_fn()
    # read variable
    num_ranges = parse(Int, readline())

    for i in 1:num_ranges
        parsed = [parse(Int, x) for x in split(readline())]
        lower = parsed[1]
        upper = parsed[2]
        solve_range(lower, upper)
    end

    # loop
    return nothing
end

main_fn()

File: hex_sum_gcd.java
/* IMPORTANT: Multiple classes and nested static classes are supported */

//imports for BufferedReader
import java.io.BufferedReader;
import java.io.InputStreamReader;

//import for Scanner and other utility classes
import java.util.Scanner;

import java.math.BigInteger;

// Warning: Printing unwanted or ill-formatted data to output will cause the test cases to fail

class TestClass {
    
    private static BigInteger hexSum(BigInteger val) {
        final BigInteger HEX_BASE = BigInteger.valueOf(16);

        BigInteger count = BigInteger.ZERO;
        for(BigInteger i = val; !i.equals(BigInteger.ZERO); i = i.divide(HEX_BASE)) {
            BigInteger hexCoeff = i.mod(HEX_BASE);
            count = count.add(hexCoeff);
        }

        return count;
    }

    private static void solve_range(BigInteger lower, BigInteger upper) {
        int non_1_gcd = 0;
        for(BigInteger j = lower; j.compareTo(upper) < 1; j = j.add(BigInteger.ONE)) {
            final BigInteger f_j = hexSum(j);
            final BigInteger hex_gcd = j.gcd(f_j);
            if(!hex_gcd.equals(BigInteger.ONE)) {
                ++non_1_gcd;
            }
        }

        //printf("%d\n", non_1_gcd);
        System.out.println(non_1_gcd);
    }

    public static void main(String args[]) throws Exception {
        final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        final String numCasesLine = br.readLine().trim();
        final int numCases = Integer.parseInt(numCasesLine);

        for(int i = 0; i < numCases; ++i) {
            String[] rangeLine = br.readLine().trim().split(" ");
            final BigInteger lower = new BigInteger(rangeLine[0]);
            final BigInteger upper = new BigInteger(rangeLine[1]);

            solve_range(lower, upper);
        }
    }
}
